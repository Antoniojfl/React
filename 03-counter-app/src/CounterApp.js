import React,{useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value})=>{

    let [count,setCount] = useState(0); //[]
    
    const handleAdd = (e) =>{
        console.log(e);
        //return ()=> console.log('Hola Mundo');
    }

    const handleSubstract = () =>{setCount(count -1);}
    
     const handleReset = () =>{setCount(value);}
    

    return (
    <>
        <h1>CounterApp</h1>
        <h3>{value}</h3>
        <h2>{count}</h2>

        <button onClick={handleAdd}>+1</button>
        <br></br>
        <button onClick={()=>setCount((count+1))}>+1 Count</button> 
        <br></br>
        <button onClick={()=>setCount((c)=>c+1)}>+1 Count otra forma</button>
        <br></br>
        <button onClick={handleSubstract}>-1 Count </button>
        <br></br>
        <button onClick={handleReset}>Reset Count </button>   
    </>);

        /*No se le pasan argumentos al on click porque al estar definido
        como funcion de flecha automaticamente se le pasan los argumentos 
        del handleAdd*/

        /*Hay que tener en cuenta que si ejecutamos la funcion handleAdd()
        react la va a ejecutar cada vez que se renderiza la pagina
        por eso hay que pasar la referencia handleAdd (sin parentesis)
        dependiendo de lo que se quiera lograr*/

        
};

CounterApp.propTypes = {
    value: PropTypes.number
};

export default CounterApp;
