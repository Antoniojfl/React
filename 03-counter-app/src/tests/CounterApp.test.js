import React from 'react';
//import {getByText, render} from '@testing-library/react'
import {shallow} from 'enzyme'
import CounterApp from '../CounterApp';



describe('Pruebas en counter app', () => {

    let wrapper= shallow(<CounterApp />);

    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    })

/*Runs a function before each of the tests in this file runs. 
If the function returns a promise or is a generator, Jest waits
for that promise to resolve before running the test.

Optionally, you can provide a timeout (in milliseconds) for specifying
how long to wait before aborting. Note: The default timeout is 5 seconds.

This is often useful if you want to reset some global state that will be
used by many tests.*/

    test('Debera de hacer match con un snapshot ', () => {

       

        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debera de devolver 100', () => {

        const wrapper = shallow(<CounterApp value ={100}/>).find('h3').text();
        //console.log(wrapper);
 
         expect(wrapper).toBe('100');
         
     });


     test('Debera de Aumentar en 1', () => {

        wrapper.find('button').at(1).simulate('click');
        const wrapperText = wrapper.find('h2').text();
 
        expect(wrapperText).toBe('1');
         
     });

     test('Debera de Disminuir en 1', () => {

        wrapper.find('button').at(3).simulate('click');
        const wrapperText = wrapper.find('h2').text();
 
        expect(wrapperText).toBe('-1');
         
     });

     test('Debera de reiniciar el valor a 0', () => {

        const wrapper = shallow(<CounterApp value ={0}/>);
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(4).simulate('click');
        const wrapperText = wrapper.find('h2').text();
        
        expect(wrapperText).toBe('0');
         
     });
     
     
}) 
