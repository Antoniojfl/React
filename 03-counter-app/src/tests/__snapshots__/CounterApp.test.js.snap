// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en counter app Debera de hacer match con un snapshot  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h3 />
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    +1 Count
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    +1 Count otra forma
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    -1 Count 
  </button>
  <br />
  <button
    onClick={[Function]}
  >
    Reset Count 
  </button>
</Fragment>
`;
